client/state/signup/progress/actions.js

```diff
@@ -55,6 +55,10 @@ function recordSubmitStep( stepName, providedDependencies, optionalProps ) {
 				propValue = otherProps;
 			}
 
+			if ( propName === 'selected_page_titles' && propValue?.length ) {
```

Comment:
One nitpick: String values also support `length` property, so `propValue?.length" will pass when it is a string, but the following `join()` will crash. Just a small detail that won't block this from shipping, of course :)

Reply:
Good catch! I have updated this PR. I have used `Array.isArray` in the fix. This also fixes the issue if the propValue is an empty array.